pkg/libzmix/bbs/bbs.go

```diff
@@ -0,0 +1,709 @@
+	}
+
+	return VerifyBlindCommitmentContext{data: handle}, nil
+}
+
+//SetPublicKey sets the bbs public key on the VerifyBlindCommitmentContext
+func (r *VerifyBlindCommitmentContext) SetPublicKey(publicKey HandleByteBuffer) error {
+	var err C.ExternError
+
+	code := C.bbs_verify_blind_commitment_context_set_public_key(r.data, publicKey.Buffer, &err)
+	if code != 0 {
+		return errors.Errorf("error occurred setting secret key to BBS sign context: %s", C.GoString(err.message))
+	}
+
+	return nil
+}
+
+//SetNonce sets the nonce on the VerifyBlindCommitmentContext
+func (r *VerifyBlindCommitmentContext) SetNonce(nonce []byte) error {
+	var err C.ExternError
+	var nonceBuffer C.ByteBuffer
+	defer C.bbs_byte_buffer_free(nonceBuffer)
+
+	nonceBuffer.len = (C.long)(len(nonce))
+	nonceBuffer.data = (*C.uchar)(unsafe.Pointer(&nonce))
+
+	code := C.bbs_verify_blind_commitment_context_set_nonce_bytes(r.data, nonceBuffer, &err)
+	if code != 0 {
+		return errors.Errorf("error occurred setting nonce bytes on BBS verify blind commitment context: %s", C.GoString(err.message))
+	}
+
+	return nil
+}
+
+//AddBlinded adds the index of the blinded message to the VerifyBlindCommitmentContext
+func (r *VerifyBlindCommitmentContext) AddBlinded(index int) error {
+	var err C.ExternError
+
+	code := C.bbs_verify_blind_commitment_context_add_blinded(r.data, (C.uint)(index), &err)
+	if code != 0 {
+		return errors.Errorf("error occurred setting blinded index on BBS verify blind commitment context: %s", C.GoString(err.message))
+	}
+
+	return nil
+}
+
+//SetProof sets the context from the BlindCommitmentContext on the VerifyBlindCommitmentContext
+func (r *VerifyBlindCommitmentContext) SetProof(proof HandleByteBuffer) error {
+	var err C.ExternError
+
+	code := C.bbs_verify_blind_commitment_context_set_proof(r.data, proof.Buffer, &err)
+	if code != 0 {
+		return errors.Errorf("error occurred setting proof on BBS verify blind commitment context: %s", C.GoString(err.message))
+	}
+
+	return nil
+}
+
+//Finish verifies the BlindCommitmentContext
+func (r *VerifyBlindCommitmentContext) Finish() error {
+	var err C.ExternError
+
+	code := C.bbs_verify_blind_commitment_context_finish(r.data, &err)
+	if code != 200 {
+		return errors.Errorf("error occurred finishing BBS verify blind commitment context: %s, %d", C.GoString(err.message), code)
+	}
+
+	return nil
+}
+
+type VerifyContext Handle
+
+//VerifyContextInit initializes a VerifyContext
+func VerifyContextInit() (VerifyContext, error) {
+	var err C.ExternError
+
+	handle := C.bbs_verify_context_init(&err)
+	if err.code != 0 {
+		return VerifyContext{}, errors.Errorf("error occurred initializing BBS verify context: %s", C.GoString(err.message))
+	}
+
+	return VerifyContext{data: handle}, nil
+}
+
+//AddMessages adds the messages to the VerifyContext
+func (r *VerifyContext) AddMessages(messages []string) error {
+	var err C.ExternError
+	var message C.ByteBuffer
+	defer C.bbs_byte_buffer_free(message)
+
```

Comment:
I don't think this free call is needed.  You are allocating a Go struct, not allowing the Rust lib to allocate the buffer.